README.md

```diff
@@ -1,22 +1,91 @@
 [![Build Status](https://travis-ci.com/HumanCompatibleAI/adversarial-policies.svg?branch=master)](https://travis-ci.com/HumanCompatibleAI/adversarial-policies)
 
-Preliminary research investigating adversarial policies: given a victim policy 
-in a multi-agent system, find a policy which will break the victim.
+Codebase to train, evaluate and analyze adversarial policies: policies attacking a victim in
+a multi-agent system. See [paper](https://arxiv.org/abs/1905.10615) for more information.
 
-# Setup
+# Installation
 
-This codebase assumes Python 3.6. Install the requirements in 
-`requirements-build.txt` before those in `requirements.txt`.
-Anaconda users can install directly with `conda env create -f environment.yml`.
+The easiest way to install the code is to build the Docker image in the `Dockerfile`.
+This will install all necessary binary and Python dependencies.
+You can also pull a Docker image for the latest master commit from
+`humancompatibleai/adversarial_policies:latest`.
+If you want to run outside of Docker (for example, for ease of development), read on.
 
-Then install *one of* `requirements-aprl.txt` or `requirements-modelfree.txt`
-depending on which experiments you need to run (unfortunately these codebases
-depend on different MuJoCo and Gym versions.)
+This codebase uses Python 3.6. The main binary dependencies are MuJoCo (version 1.3.1 for
+`modelfree` and 1.5.0 for `aprl`). You may also need to install some other libraries,
+such as OpenMPI.
 
-Finally, `pip install -e .` to install an editable version of the package.
+There are two different parts of the codebase, `src/aprl` and `src/modelfree` with mutually
+exclusive requirements (due to different Gym and MuJoCo versions). You can create a
+virtual environment for each by running `ci/build_venv.sh` with either `aprl` or `modelfree`.
+Finally, inside the relevant virtual environment run `pip install -e .` to install an editable
+version of the package.
+
+# Reproducing Results
+
+## Training adversarial policies
+
+`modelfree.train` trains a single adversarial policy. `modelfree.multi.train` trains multiple
+adversarial policies, using Ray (see below) for parallelization.
```

Comment:
Could be worth also mentioning that we use Sacred for experiment setup, and linking to it, since that might give people signposts for parsing what's being captured in the output directories.

Reply:
thanks, added